FortranRuntime/CMakeLists.txt

```diff
@@ -0,0 +1,161 @@
+#===-- CMakeLists.txt ------------------------------------------------------===#
```

Comment:
I used FortranRuntime so far because the library's name is `libFortranRuntime.a`, so it is the path least amount of changes and I myself don't have a favorite alternative which I would strongly prefer. This PR would be the occasion to discuss a new name.

https://reviews.llvm.org/D154869 suggested `flang-rt` as folder name and `libflang-rt.a` as library name. This patch puts the library next to `libclang_rt.*.a` from `compiler-rt` so that seems to be a naming scheme with precedence although inconsistent hyphenation.

So let me start with a suggestion: `flang-rt` as top-level folder name, `libflang_rt.a` as library name, as `flang-rt/{lib,include/flang_rt}/Runtime` for the source files.

Reply:
> So let me start with a suggestion: `flang-rt` as top-level folder name, `libflang_rt.a` as library name, as `flang-rt/{lib,include/flang_rt}/Runtime` for the source files.

I don't have a vote, but that sounds like a good plan. It also matches the top-level name of the reverted-and-not-brought-back https://github.com/llvm/llvm-project/commit/6403287eff71a3d6f6c862346d6ed3f0f000eb70